File: src/data_structures/queues/Queues.kt
package data_structures.queues

class Queues<T> {
    private val list = arrayListOf<T>()
    val count: Int
        get() = list.size
    val isEmpty: Boolean
        get() = list.isEmpty()

    fun peek(): T? = list.getOrNull(0)

    fun enqueue(element: T): Boolean {
        list.add(element)
        return true
    }

    fun dequeue(): T? = if (isEmpty) null else list.removeAt(0)

    override fun toString(): String = buildString {
        append("- front - ")
        list.forEach {
            append("| $it ")
        }
        append(" - back -")
    }
}

File: data_structures/queues/queues.py
class Queue(object):
    def __init__(self):
        self.list = list()

    def is_empty(self):
        return len(self.list) == 0

    def enqueue(self, element):
        self.list.append(element)
        return True
    
    def dequeue(self):
        if self.is_empty():
            return None
        return self.list.pop(0)

    def to_string(self):
        str = 'font-|'
        for e in self.list:
            str+='{0}|'.format(e)
        str += '-back'
        return str
